daemon/cmd/daemon.go

```diff
@@ -683,8 +685,9 @@ func newDaemon(ctx context.Context, cleaner *daemonCleanup,
 
 	d.cgroupManager = manager.NewCgroupManager()
 
-	if option.Config.EnableIPv4EgressGateway {
-		d.egressGatewayManager = egressgateway.NewEgressGatewayManager(cacheStatus, d.identityAllocator, option.Config.InstallEgressGatewayRoutes)
+	var egressGatewayWatcher watchers.EgressGatewayManager
+	if d.egressGatewayManager != nil {
+		egressGatewayWatcher = d.egressGatewayManager
```

Comment:
Can't we just pass `d.egressGatewayManager` to `watchers.NewK8sWatcher()`?

Reply:
Unfortunately not, because of the dreaded `nil` interface bug. Passing directly will construct the following:

```go
var v watchers.EgressGatewayManager = (*Manager)(nil)
// we need
var v watchers.EgressGatewayManager = nil
```

Want me to comment that?

Reply:
Makes sense, no need to add a comment